### ISSUE TITLE
In headless mode NodeRunner passes --no-local-shell to WebServer causing it to fail

### ISSUE THREAD

**Issue by george-hawkins** [corda/corda-gradle-plugins#12]
When you use `runnodes` this invokes [`NodeRunner`](cordformation/src/noderunner/kotlin/net/corda/plugins/NodeRunner.kt) and if the environment is headless or you specify `--headless` the it uses [`HeadlessJavaCommand`](https://github.com/corda/corda-gradle-plugins/blob/master/cordformation/src/noderunner/kotlin/net/corda/plugins/NodeRunner.kt#L113) which causes the executable jars that are found to all be started with the additional argument `--no-local-shell`.

The [`Corda`](https://github.com/corda/corda/blob/master/node/src/main/kotlin/net/corda/node/Corda.kt) nodes know how to consume this argument as they use an [`ArgParser`](https://github.com/corda/corda/blob/master/node/src/main/kotlin/net/corda/node/ArgsParser.kt#L38) that expects this.

However the [`WebServer`](https://github.com/corda/corda/blob/master/webserver/src/main/kotlin/net/corda/webserver/WebServer.kt) processes choke on this argument as they use a different [`ArgParser`](https://github.com/corda/corda/blob/master/webserver/src/main/kotlin/net/corda/webserver/WebArgsParser.kt#L21) that doesn't know what to do with this argument.

So when you use `runnode` you see something like this:

```
$ ./runnodes
Starting nodes in /vagrant
Starting corda.jar in /vagrant/PartyB on debug port 5005
Starting corda-webserver.jar in /vagrant/PartyB on debug port 5006
Started 2 processes
Finished starting nodes
Listening for transport dt_socket at address: 5006
Unknown command line arguments: no-local-shell is not a recognized optionUnknown command line arguments: no-local-shell is not a recognized option
```

I.e. here you see it starting both a Corda node and a webserver process but the webserver quickly exits with the `Unknown command line arguments: no-local-shell` error.

This means the introductory [corda app](https://docs.corda.net/tutorial-cordapp.html) fails fairly mysteriously if you try running it on e.g. a headless VM (as I guess is typical with people using things like Docker or, in my case, Vagrant).

**Comment by george-hawkins** [comment]
The push I've submitted applies to the new separate corda-gradle-plugins project.

If you want you can apply the same change to the plugin as it existed for Corda v2.0 (when it was still part of the main [corda project](https://github.com/corda/corda)) with the following steps.

**1.** First download my patch that's suitable for v2.0:

    $ cd ~/Downloads
    $ curl -O https://gist.githubusercontent.com/george-hawkins/0d5a33df099f990e2ba1b7dc06a3f47b/raw/71c5799569b7348a5df83249df58f7e0eebe9d20/corda-v2.0-gradle-plugin.patch

**2.** Then clone the Corda repo, check out v2.0, apply the patch and build:

    $ git clone https://github.com/corda/corda.git
    $ cd corda
    $ git checkout -b release-V2.0 release-V2.0
    $ git apply ~/Downloads/corda-v2.0-gradle-plugin.patch
    $ cd gradle-plugins
    $ ../gradlew build
    $ cp cordformation/build/libs/cordformation-1.0.2-fatjar.jar runnodes.jar

With the new `runnodes.jar`, that you now have, you have a version of the jar that can be used to successfully start the web server that accompanies nodes, e.g. as found in the introductory [corda app](https://docs.corda.net/tutorial-cordapp.html), in headless environments.

**Comment by mikehearn** [comment]
@andr3ej has encountered the same issue (or seen this one) and fixed this today.

**Comment by mikehearn** [comment]
I believe this is fixed in plugins version 3.0.9 (which requires Corda v3 which isn't out yet). I'll mark it as fixed for now. If I'm wrong @andr3ej can reopen.
